### DISCUSSION TITLE
How can you use the modulo operator (%) to determine if a number is even or odd?

### DISCUSSION THREAD

**Discussion by yawarasuuna** [Cyfrin/foundry-full-course-cu#2768] [Q&A] [answered]
One of the question on Foundry Fundamentals, Section 4.11 for "Quiz: Smart Contract Lottery Intro", it asks "How can you use the modulo operator (%) to determine if a number is even or odd?"

![Screenshot 2024-10-28 at 15-38-07 Quiz Smart Contract Lottery Intro Cyfrin Updraft](https://github.com/user-attachments/assets/4e381de0-e2c2-4ce1-a83e-9af5d8d95986)

Wouldn't both answers B and C allow us to determine if a number is even or odd?

```
if n % 2 != 1 // If n is not odd ( n is even), otherwise it's odd
```

```
if n % 2 == 0 // If n is even, otherwise it's odd
```

Sorry if it was already asked, but I did not find anything similar to this case, and I'm not sure if AI might be hallucinating with me lol

Thank you,

**Comment by EngrPips** [comment]
Seem to me that `B` and `C` are both correct but maybe I am wrong.

**Reply by BowtiedHarpyEagle** [reply, marked as answer]
Yeah, it is weird that this is a question with 2 correct answers. However, C seems to be more direct, and B seems to be technically correct but a bit more indirect and less readable. But yeah, both would work.

**Reply by EngrPips** [reply]
This is an interesting comment.
